espresso/streamer.go

```diff
@@ -299,3 +312,18 @@ func (s *EspressoStreamer[B]) HasNext(ctx context.Context) bool {
 
 	return false
 }
+
+// This function allows to "pin" the Espresso block height corresponding to the last safe batch
```

Comment:
I understand that this is where [your algorithm](https://app.asana.com/1/1208976916964769/project/1209392461754458/task/1210076733495976?focus=true) for resetting the streamer's HotShot block height is handled. Please add a few comments and a reference to the book as this is non-trivial.